parallel-install/pkg/deployment/coredns.go

```diff
@@ -0,0 +1,152 @@
+package deployment
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"html/template"
+	"strings"
+
+	"github.com/avast/retry-go"
+	"github.com/kyma-incubator/hydroform/parallel-install/pkg/logger"
+	v1 "k8s.io/api/core/v1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+var coreDNSPatchTemplate = `
+.:53 {
+    errors
+    health
+    rewrite name regex {{ .DomainName}} istio-ingressgateway.istio-system.svc.cluster.local
+    ready
+    kubernetes cluster.local in-addr.arpa ip6.arpa {
+      pods insecure
+      fallthrough in-addr.arpa ip6.arpa
+    }
+    hosts /etc/coredns/NodeHosts {
+      reload 1s
+      fallthrough
+    }
+    prometheus :9153
+    forward . /etc/resolv.conf
+    cache 30
+    loop
+    reload
+    loadbalance
+}
+`
+
+// CoreDNSPatch contains values to fill template with
+type CoreDNSPatch struct {
```

Comment:
Btw - do we need a separate type for this patch? It holds only one value.

Reply:
I think the type adds nice context to the value, so it serves the documentation purpose better than a variable name or  a comment. It's OK to have it IMO.

Reply:
Agree. 👍

Reply:
We need that in order to fill golang template with some variables, `domainName` in this case. I don't know if there is another way for doing that.